apps/wallet/src/ui/app/pages/home/transactions/index.tsx

```diff
@@ -58,4 +57,4 @@ function TransactionBlocksPage() {
     );
 }
 
-export default memo(TransactionBlocksPage);
+export default TransactionBlocksPage;
```

Comment:
I'm guessing `memo` was the issue because we have to re-create the whole component tree whenever the network changes due to `api.instance.fullNode` being a global, and so `memo` was preventing this component from being re-rendered? if so, this is tricky and I wish we could fix the root issue to prevent subtle bugs like this in the future  😅

Reply:
@williamrobertson13 this hook `useQueryTransactionsByAddress` never resolves the loading state back to `false`. I am trying to see if I can do anything else besides that... But for now, removing `memo` forces that hook that re-run and return correctly.

Reply:
There are no props, and the `activeAddress` is also not changed... so the whole thing just does not get re-run.

Reply:
FWIW I want us to _really_ avoid using `memo` / `useMemo` because I feel like they are generally mis-understood.